File: src/main/java/com/googongill/aditory/service/dto/link/LinkResult.java
package com.googongill.aditory.service.dto.link;

import com.googongill.aditory.domain.Category;
import com.googongill.aditory.domain.Link;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class LinkResult {
    private Long linkId;
    private Long categoryId;
    private LocalDateTime createdAt;
    private LocalDateTime lastModifiedAt;

    public static LinkResult of(Link link, Category category) {
        return LinkResult.builder()
                .linkId(link.getId())
                .categoryId(category.getId())
                .createdAt(link.getCreatedAt())
                .lastModifiedAt(link.getLastModifiedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/link/LinkResult.kt
package com.googongill.aditory.service.dto.link

import com.googongill.aditory.domain.Category
import com.googongill.aditory.domain.Link
import java.time.LocalDateTime

data class LinkResult(
    val linkId: Long?,
    val categoryId: Long?,
    val createdAt: LocalDateTime?,
    val lastModifiedAt: LocalDateTime?
) {

    companion object {
        fun of(link: Link, category: Category): LinkResult {
            return LinkResult(
                linkId = link.id,
                categoryId = category.id,
                createdAt = link.createdAt,
                lastModifiedAt = link.lastModifiedAt
            )
        }
    }
}
